### DISCUSSION TITLE
How Do I Persist Cache With Docker?

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#36935] [Help] [answered]
Hi,

I'm trying to persist Nextjs cache across rebuilds of my docker container. I have added the line `VOLUME /app/.next/cache` to my Dockerfile but the volume doesn't appear to be being used by Next (The volume has been created). My WORKDIR is /app.

Could anyone please offer any advice on how to achieve a persistent cache?

Update, I Have also tried with `VOLUME ["/app/.next/cache"]`

**Comment by darrylmorley** [comment, marked as answer]
I found the answer, this is what I needed: 
```
RUN mkdir -p /app/.next/cache && chown nextjs:nodejs /app/.next/cache
VOLUME ["/app/.next/cache"]
```

**Reply by deadcoder0904** [reply]
@darrylmorley does this work for development build or production build too? because for some reason, my `next build` always takes time like 30+ seconds even after i added your code. [my dockerfile for example.](https://github.com/vercel/next.js/discussions/35296#discussioncomment-8626231)

**Comment by alexschneider-dev** [comment]
If you use VOLUME ["/app/.next/cache"] in your Dockerfile, anytime a new container is created, it will create a new volume.
If you are using a VPS, It is better to create volume on that VPS, and specify the mount source either in the docker-compose.yml or with the run command. I use the run command, so in my case:

docker run --name nextjs -d
-p 3000:3000
-p 3001:3001
--network=nextjs
--mount source=nextjs_cache,target=/app/.next

**Reply by darrylmorley** [reply]
Thanks for adding this.

**Comment by alexschneider-dev** [comment]
Updated Sep 18 2024, 

if you do it like in my previous comment, i.e. mounting the whole .next folder, it will cache everything. If you add any changes to the codebase and deploy a new container, it will serve the old cached version of a page from the mounted volume. I fixed this problem by only mounting the Images folder in the .next folder.

docker stop nextjs && \
docker rm nextjs && \
docker run --name nextjs -d \
  -p 3000:3000 \
  -p 3001:3001 \
  --network=nextjs \
  --mount source=nextjs_cache,target=/app/.next/cache/images \
  <your docker container from repo>
